xdsl/dialects/onnx.py

```diff
@@ -134,6 +137,124 @@ def verify_(self) -> None:
                 raise VerifyException("Mismatch between operand type and res type")
 
 
+@irdl_op_definition
+class Gemm(IRDLOperation):
+    name = "onnx.Gemm"
+    """
+    General Matrix multiplication: https://en.wikipedia.org/wiki/Basic_Linear_Algebra_Subprograms#Level_3
+    A' = transpose(A) if transA else A
+    B' = transpose(B) if transB else B
+    Compute Y = alpha * A' * B' + beta * C,
+    where input tensor A has shape (M, K) or (K, M), input tensor B has shape (K, N) or (N, K),
+    input tensor C is broadcastable to shape (M, N), and output tensor Y has shape (M, N).
+    """
+    T = Annotated[AnyFloat | IntegerType, ConstraintVar("T")]
+    tensor_a = operand_def(TensorType[T])
+    tensor_b = operand_def(TensorType[T])
+    tensor_c = operand_def(TensorType[T])
+
+    alpha = opt_attr_def(FloatAttr)
+    beta = opt_attr_def(FloatAttr)
+
+    trans_a = opt_attr_def(IntegerAttr)
+    trans_b = opt_attr_def(IntegerAttr)
```

Comment:
```suggestion
    trans_a = opt_attr_def(IntegerAttr, attrname="transA")
    trans_b = opt_attr_def(IntegerAttr, attrname="transB")
```
This bridges our Python code style to the naming of the attribute names in the dialect, without it the dict below won't work